org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/uriresolver/InvalidURIException.java

```diff
@@ -0,0 +1,33 @@
+/**
+ *  Copyright (c) 2022 Red Hat Inc. and others.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v2.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ *  Contributors:
+ *  Red Hat Inc. - initial API and implementation
+ */
+package org.eclipse.lemminx.uriresolver;
+
+/**
+ * Invalid URI exception.
+ *
+ * @author Fred Bricon
+ *
+ */
+public class InvalidURIException extends CacheResourceException {
+
+	private static final long serialVersionUID = 1L;
+	
+	public InvalidURIException(String resourceURI, String message, Throwable cause) {
```

Comment:
To be consistent with cacheresourcedownloadingewception the message should be in this class